coalib/output/Logging.py

```diff
@@ -1,10 +1,26 @@
 from datetime import datetime
+from collections import defaultdict
 import json
 import io
 import logging
 import logging.config
 
 
+class CounterHandler(logging.Handler):
+    """
+    A logging handler which counts the number of calls
+    for each logging level.
+    """
+    level_called = None
+
+    def __init__(self, *args, **kwargs):
+        super(CounterHandler, self).__init__(*args, **kwargs)
+        CounterHandler.level_called = defaultdict(lambda: 0)
+
+    def emit(self, record):
+        CounterHandler.level_called[record.levelname] += 1
```

Comment:
Better use `type(self)` instead of `CounterHandler`. So you get independent from the class name.

Or even more better ;) Turn `emit` into:

```python
@classmethod
def emit(cls, record):
    cls.level_called[record.levelname] += 1
```